controllers/configurationpolicy_controller.go

```diff
@@ -2556,6 +2558,29 @@ func handleSingleKey(
 			mergedValue.(map[string]interface{}), existingValue.(map[string]interface{}))
 	}
 
+	if key == "stringData" && existingObj.UnstructuredContent()["kind"] == "Secret" {
+		// override automatic conversion from stringData to data prior to evaluation
+		existingValue = existingObj.UnstructuredContent()["data"]
+		value, ok := existingValue.(map[string]interface{})
+
+		if !ok {
+			message := "Error casting encoded data"
+
+			return message, false, mergedValue, false
+		}
+
+		for k, val := range value {
+			decodedVal, err := base64.StdEncoding.DecodeString(val.(string))
+			if err != nil {
+				message := fmt.Sprintf("Error decoding secret: %s", k)
```

Comment:
This message sounds like it will include the secret's name, but actually that's the key in the secret that couldn't be decoded... is that right?